### PR TITLE
experiment with github actions

### PR DESCRIPTION
**Pull request by mroz22**
experimenting with gh actions to do some preliminary checks for external contributorrs pull requests. I am not experienced with gh actions so happy to hear others input.

### CODE REVIEW

File: .github/workflows/validation.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      - name: Type check
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12.16.3
+      - run: npm install -g yarn
+      - name: yarn install, build and test
+        run: |
+          yarn
+          yarn build:libs
+          yarn workspace @trezor/suite type-check
```
Comment (szymonlesisz) [inline .github/workflows/validation.yml]:
```suggestion
          yarn type-check
          yarn lint
```
packages other than suite might be affected as well, no?

File: ci/build.yml [hunk 2]
```diff
@@ -118,7 +118,7 @@ suite-web-landing build stable:
     expire_in: 1 days
 
 .build_nix: &build_nix
-  script: # override build script to use nix-shell instead
+  script:  # override build script to use nix-shell instead
```
Comment (szymonlesisz) [inline ci/build.yml:121]: space is neccessary? same below

Comment (mroz22) [reply ci/build.yml:121]: not really but yaml linter issues warning over this missing space https://gitlab.com/satoshilabs/trezor/trezor-suite/-/jobs/1615420866

Comment (szymonlesisz) [reply ci/build.yml:121]: ah, cool

**Comment by szymonlesisz** [comment]
any idea why this new job was not triggered?

**Comment by mroz22** [comment]
> any idea why this new job was not triggered?

no idea, I will try to re-create this PR from another branch.

edit: did not help

**Comment by mroz22** [comment]
image
@vdovhanych any idea why is there this exclamation mark and clicking it opens generic github 500 error page?

**Comment by szymonlesisz** [comment]
maybe it should be?
```
on:
  pull_request:
    branches: [develop]
```

**Comment by mroz22** [comment]
ok, so the problem was invalid gh actions syntax. Github UI is not very intuitive here but I was able to navigate to the log finally :D 
https://github.com/trezor/trezor-suite/runs/3711868055?check_suite_focus=true

**Review by matejkriz — APPROVED** [review]

**Pull request merged**